go/libkb/naclwrap.go

```diff
@@ -680,6 +691,19 @@ func (k NaclDHKeyPair) EncryptToString(plaintext []byte, sender GenericKey) (str
 	return PacketArmoredEncode(info)
 }
 
+func (k NaclDHKeyPair) SecretSymmetricKey(reason EncryptionReason) ([]byte, error) {
+
+	if !k.CanDecrypt() {
+		return nil, KeyCannotDecryptError{}
+	}
```

Comment:
To sanity check, should we also require that `len(reason) > 8` or so?  Just to make sure we're not passing something silly in here.